base_layer/core/src/transactions/transaction.rs

```diff
@@ -244,18 +245,27 @@ impl UnblindedOutput {
     /// Commits an UnblindedOutput into a Transaction input
     pub fn as_transaction_input(&self, factory: &CommitmentFactory) -> Result<TransactionInput, TransactionError> {
         let commitment = factory.commit(&self.spending_key, &self.value.into());
-        let script_nonce = PrivateKey::random(&mut OsRng);
-        let public_script_nonce = PublicKey::from_secret_key(&script_nonce);
+        let script_nonce_a = PrivateKey::random(&mut OsRng);
+        let script_nonce_b = PrivateKey::random(&mut OsRng);
+        let nonce_commitment = factory.commit(&script_nonce_b, &script_nonce_a);
+
         let e = Challenge::new()
-            .chain(public_script_nonce.as_bytes())
+            .chain(nonce_commitment.as_bytes())
             .chain(self.script.as_bytes().as_slice())
             .chain(self.input_data.as_bytes().as_slice())
-            //.chain(&self.height.to_le_bytes()) //TODO decide if the height should remain in this signature
+            .chain(PublicKey::from_secret_key(&self.script_private_key).as_bytes())
+            .chain(commitment.as_bytes())
             .result()
             .to_vec();
-
-        let script_signature = Signature::sign(self.script_private_key.clone(), script_nonce, &e)
-            .map_err(|_| TransactionError::InvalidSignatureError("Generating script signature".to_string()))?;
+        let script_signature = ComSignature::sign(
+            self.value.into(),
+            self.script_private_key.clone() + self.spending_key.clone(),
```

Comment:
Are `script_private_key` and `spending_key` always generated by the same individual? If not, we've got to be very careful we're not introducing any opportunities here for rogue key attacks.

Reply:
If we are sticking with MW rules, yes. This is specifically to force the user who generated both to be the same individual, as there are attacks that someone not being the spender be able to change the data here. 

For example:
Alice spends her UTXO with a NOP script on it, which allows any input on it. Basically mimicking vanilla MW behavior. Bon can then take Alice's input and change the input data and thus change the `script_private_key`. If Bob cab changes this, he can the spending conditions on his utxo.  This is discussed a bit more here in detail: https://github.com/tari-project/tari/blob/tari-script/RFC/src/RFC-0201_TariScript.md#script-offset-security